zipline/sources/data_frame_source.py

```diff
@@ -108,11 +119,21 @@ class DataPanelSource(DataSource):
     def __init__(self, data, **kwargs):
         assert isinstance(data.major_axis, pd.tseries.index.DatetimeIndex)
 
-        self.data = data
+        self.data = data.fillna(method='ffill', axis=0)
```

Comment:
Why are we `'fillna'`ing everything here?  We probably don't want to be ffilling anything except price.  Also from testing in the shell I don't think passing an axis to ffill does anything for a panel.